dali/operators/reader/video_reader_decoder_op_test.cc

```diff
@@ -0,0 +1,299 @@
+    .AddOutput("frames", "cpu"));
+
+  pipe.Build({{"frames", "cpu"}});
+
+  std::vector<int> expected_order = {44, 12, 49, 38, 8};
+
+  int num_sequences = 5;
+
+  for (int sequence_id = 0; sequence_id < num_sequences; ++sequence_id) {
+    DeviceWorkspace ws;
+    pipe.RunCPU();
+    pipe.RunGPU();
+    pipe.Outputs(&ws);
+
+    auto &frame_video_output = ws.template OutputRef<dali::CPUBackend>(0);
+    const auto sample = frame_video_output.tensor<uint8_t>(0);
+    ComapreFrames(sample, this->GetCfrFrame(0, expected_order[sequence_id]), this->FrameSize(0));
+  }
+}
+
+
+TEST_F(VideoReaderDecoderTest, CompareReaders) {
+  const int batch_size = 4;
+  const int sequence_length = 6;
+  const int stride = 3;
+  const int step = 10;
+
+  Pipeline pipe(batch_size, 4, 0);
+
+  pipe.AddOperator(OpSpec("experimental__readers__Video")
+    .AddArg("device", "cpu")
+    .AddArg("sequence_length", sequence_length)
+    .AddArg("stride", stride)
+    .AddArg("step", step)
+    .AddArg(
+      "filenames",
+      std::vector<std::string>{
+        testing::dali_extra_path() + "/db/video/cfr/test_1.mp4",
+        testing::dali_extra_path() + "/db/video/cfr/test_2.mp4"})
+    .AddArg("labels", std::vector<int>{0, 1})
+    .AddOutput("frames", "cpu")
+    .AddOutput("labels", "cpu"));
+  pipe.AddOperator(OpSpec("readers__Video")
+    .AddArg("device", "gpu")
+    .AddArg("sequence_length", sequence_length)
+    .AddArg("stride", stride)
+    .AddArg("step", step)
+    .AddArg(
+      "filenames",
+      std::vector<std::string>{
+        testing::dali_extra_path() + "/db/video/cfr/test_1.mp4",
+        testing::dali_extra_path() + "/db/video/cfr/test_2.mp4"})
+    .AddArg("labels", std::vector<int>{0, 1})
+    .AddOutput("frames_gpu", "gpu")
+    .AddOutput("labels_gpu", "gpu"));
+
+  pipe.Build({{"frames", "cpu"}, {"frames_gpu", "gpu"}, {"labels", "cpu"}, {"labels_gpu", "gpu"}});
+
+  int num_sequences = 20;
+  int sequence_id = 0;
+  int batch_id = 0;
+  int gt_frame_id = 0;
+
+  int video_idx = 0;
+
+  while (sequence_id < num_sequences) {
+    DeviceWorkspace ws;
+    pipe.RunCPU();
+    pipe.RunGPU();
+    pipe.Outputs(&ws);
+
+    auto &frame_video_output = ws.template OutputRef<dali::CPUBackend>(0);
+    auto &frame_gpu_video_output = ws.template OutputRef<dali::GPUBackend>(1);
+
+    auto &frame_label_output = ws.template OutputRef<dali::CPUBackend>(2);
+    auto &frame_gpu_label_output = ws.template OutputRef<dali::GPUBackend>(3);
+
+    for (int sample_id = 0; sample_id < batch_size; ++sample_id) {
+      const auto sample = frame_video_output.tensor<uint8_t>(sample_id);
+      const auto sample_gpu = frame_gpu_video_output.tensor<uint8_t>(sample_id);
+
+      const auto label = frame_label_output.tensor<int>(sample_id);
+      const auto label_gpu = frame_label_output.tensor<int>(sample_id);
+
+      vector<uint8_t> frame_gpu(this->FrameSize(video_idx));
+
+      int label_gpu_out = -1;
+      MemCopy(&label_gpu_out, label_gpu, sizeof(int));
```

Comment:
```suggestion
      int label_gpu_out = *label_gpu;
```
would be a bit simpler, I think.

Reply:
There was a mistake above. This should have been taken from GPU output, so I need to copy it to the CPU to compare. Fixed the issue.